### PR TITLE
[release/2.7 backport] remove github.com/dgrijalva/jwt-go

### PR DESCRIPTION
**Pull request by brackendawson**
By updating github.com/Azure/go-autorest/autorest to v.0.11.19 (10e0b31633f168ce1a329dcbdd0ab9842e533fb5)

Backport of distribution/distribution#3459

#3361

### CODE REVIEW

**Review by joaodrp — APPROVED** [review]

File: vendor.conf [hunk 1]
```diff
@@ -8,9 +8,9 @@ github.com/bugsnag/bugsnag-go b1d153021fcd90ca3f080db36bec96dc690fb274
 github.com/bugsnag/osext 0dd3f918b21bec95ace9dc86c7e70266cfc5c702
 github.com/bugsnag/panicwrap e2c28503fcd0675329da73bf48b33404db873782
 github.com/denverdino/aliyungo afedced274aa9a7fcdd47ac97018f0f8db4e5de2
-github.com/dgrijalva/jwt-go a601269ab70c205d26370c16f7c81e9017c14e04
```
Comment (thaJeztah) [inline vendor.conf:11]: I see updating Azure/go-autorest brings a substantial number of code changes (for a patch release).

If the bug is in this library, could we instead update just this library to a version with the fix? I see maintains a fork with the fix (IIUC), and we can specify it with a custom location (the equivalent to `replace` in `go.mod`);

```suggestion
github.com/dgrijalva/jwt-go a211650c6ae1cff6d7347d3e24070d65dcfb1122 https://github.com/form3tech-oss/jwt-go.git # v3.2.4
```

That would only bring the diff of the jwt-go package;
https://github.com/form3tech-oss/jwt-go/compare/a601269ab70c205d26370c16f7c81e9017c14e04...v3.2.4

Comment (thaJeztah) [reply vendor.conf:11]: gave it a quick attempt; https://github.com/distribution/distribution/pull/3466

File: vendor.conf [hunk 2]
```diff
@@ -8,9 +8,9 @@ github.com/bugsnag/bugsnag-go b1d153021fcd90ca3f080db36bec96dc690fb274
...
 github.com/docker/go-metrics 399ea8c73916000c64c2c76e8da00ca82f8387ab
 github.com/docker/libtrust fa567046d9b14f6aa788882a950d69651d230b21
+github.com/form3tech-oss/jwt-go 9162a5abdbc046b7c8b03ee90052cee67e25caa7
```
Comment (thaJeztah) [inline vendor.conf:13]: This commit looks to be matching v3.2.2; https://github.com/form3tech-oss/jwt-go/compare/9162a5abdbc046b7c8b03ee90052cee67e25caa7...v3.2.2, which is missing a security fix in v3.2.4; https://github.com/form3tech-oss/jwt-go/compare/v3.2.2...v3.2.4 (see https://github.com/form3tech-oss/jwt-go/tree/v3.2.4)

File: vendor.conf (hunk 1 above)
Comment (brackendawson) [reply vendor.conf:11]: Not against using a replace if vndr can do that. Should we consider using the fork which the original repository now links to? https://github.com/golang-jwt/jwt

Comment (thaJeztah) [reply vendor.conf:11]: Oh! Arf... there's _two_ forks now, and both being actively maintained? 😞

Comment (thaJeztah) [reply vendor.conf:11]: I updated https://github.com/distribution/distribution/pull/3466 with a commit (allowing the differences between those forks to be reviewed)

**Comment by milosgajdos** [comment]
If we want this fix in 2.7.x, should we just propagate #3138 i.e. update Azure autorest to ``v0.11.20` and be done with this? If we do do that, we should then close #3466 🤔

**Comment by thaJeztah** [comment]
https://github.com/distribution/distribution/pull/3138 would bring in an even larger diff.

Screenshot 2021-09-05 at 20 46 26


The security fix is only a few line changes (which is why I opened https://github.com/distribution/distribution/pull/3466)

**Comment by milosgajdos** [comment]
> which is why I opened #3466

Oh, right. There have been a few related PRs flying around this issue that I've completely lost the track.
I say let's merge #3466. We probably don't need it per se, but if there is nothing that prevents us from having a peace of mind I see no problem with doing that 🤔

**Comment by milosgajdos** [comment]
Let's close this one in favour of #3466

**Comment by thaJeztah** [comment]
looks like github didn't auto-close this one for some reason. https://github.com/distribution/distribution/pull/3466 was merged, so let me close this one (thanks @brackendawson !)

**Pull request closed without merging**